### PR TITLE
Zerobounce integrated

### PR DESCRIPTION
**Pull request by mustafapc19**
POC for [this](https://github.com/bigbinary/neeto_emails/issues/89#issuecomment-1087806373)

- Converted mail interceptor to engine due to zero bounce configuration.
- `MailInterceptor` has a `enable_zerobounce_validation` config for optional zerobounce email validation.

[Demo](https://www.loom.com/share/79d09933170d4efd8661e8299c1a4bd3)
[Demo video with neetoForm](https://www.loom.com/share/8dbac21a538f48298f004ea9b1e3c23b)

### CODE REVIEW

**Comment by mustafapc19** [comment]
cc @ghousemohamed

File: lib/mail_interceptor/initializers/zerobounce.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+require "zerobounce"
+
+Zerobounce.configure do |config|
+  config.apikey = ENV['ZEROBOUNCE_API_KEY']
```
Comment (ghousemohamed) [inline lib/mail_interceptor/initializers/zerobounce.rb:4]: @mustafapc19 Can make this `api_key`?

**Review by ghousemohamed — CHANGES_REQUESTED** [review]
@mustafapc19 _a

**Comment by ghousemohamed** [comment]
@mustafapc19 Also during development we do not want to add this check with Zerobounce API

**Comment by mustafapc19** [comment]
> @mustafapc19 Also during development we do not want to add this check with Zerobounce API

@ghousemohamed Check with zero bounce is disabled by default. So I thought only enabling it in when environment is production would be enough. I hope I am not missing something.

File: lib/mail_interceptor/initializers/zerobounce.rb (hunk 1 above)
Comment (mustafapc19) [reply lib/mail_interceptor/initializers/zerobounce.rb:4]: @ghousemohamed I looked into `Zerobounce` codebase they seem to be expecting and using `api_key`.

**Comment by ghousemohamed** [comment]
@mustafapc19 Yes, that makes sense, we will anyway be adding this initializer only for production environments.

**Review by ghousemohamed — APPROVED** [review]

**Pull request merged**